### ISSUE TITLE
Panic when maximizing window

### ISSUE THREAD

**Issue by viperperidot** [achannarasappa/ticker#147]
When I start ticker and then maximize the window I get this panic:

```
Caught panic:

runtime error: slice bounds out of range [:-1]

mk@mk-Inspiron-15-3567:~$ 

goroutine 1 [running]:
runtime/debug.Stack(0x56, 0x0, 0x0)
	/opt/hostedtoolcache/go/1.15.8/x64/src/runtime/debug/stack.go:24 +0x9f
runtime/debug.PrintStack()
	/opt/hostedtoolcache/go/1.15.8/x64/src/runtime/debug/stack.go:16 +0x25
github.com/charmbracelet/bubbletea.(*Program).Start.func1(0xc0001dd290)
	/home/runner/go/pkg/mod/github.com/charmbracelet/bubbletea@v0.12.2/tea.go:119 +0xd3
panic(0x960a40, 0xc000341420)
	/opt/hostedtoolcache/go/1.15.8/x64/src/runtime/panic.go:969 +0x1b9
github.com/achannarasappa/term-grid.getLineText(0x0, 0xc0003998a0, 0x1, 0x1, 0x1, 0xffffffffffffffff, 0x0, 0x1, 0x1)
	/home/runner/go/pkg/mod/github.com/achannarasappa/term-grid@v0.2.0/cell.go:40 +0x251
github.com/achannarasappa/term-grid.getLines(0x0, 0x0, 0xffffffffffffffff, 0x0, 0x0, 0x0, 0xc0002d83e0, 0x2, 0x2, 0x2, ...)
	/home/runner/go/pkg/mod/github.com/achannarasappa/term-grid@v0.2.0/cell.go:74 +0x2d7
github.com/achannarasappa/term-grid.renderRow(0x96, 0xc00024e000, 0xb, 0xc, 0x1, 0xc000046400, 0x0)
	/home/runner/go/pkg/mod/github.com/achannarasappa/term-grid@v0.2.0/row.go:81 +0x1ad
github.com/achannarasappa/term-grid.Render(0xc000258000, 0xf, 0x10, 0x0, 0x1, 0xc0001dcff0, 0x9c4b18)
	/home/runner/go/pkg/mod/github.com/achannarasappa/term-grid@v0.2.0/grid.go:29 +0xcc
github.com/achannarasappa/ticker/internal/ui/component/watchlist.Model.View(0x96, 0xc000155500, 0x5, 0x8, 0xc0001dd8c0, 0x10101, 0x9c4b00, 0x5, 0x0, 0x0, ...)
	/home/runner/work/ticker/ticker/internal/ui/component/watchlist/watchlist.go:85 +0x33a
github.com/achannarasappa/ticker/internal/ui.Model.Update(0x5, 0x0, 0x0, 0x0, 0xc000156900, 0x8, 0x8, 0x101010101, 0x0, 0x0, ...)
	/home/runner/work/ticker/ticker/internal/ui/ui.go:114 +0x765
github.com/charmbracelet/bubbletea.(*Program).Start(0xc0001dd290, 0x0, 0x0)
	/home/runner/go/pkg/mod/github.com/charmbracelet/bubbletea@v0.12.2/tea.go:214 +0x61c
github.com/achannarasappa/ticker/internal/ui.Start.func1(0xc0001dcb70, 0xc0001dcc90)
	/home/runner/work/ticker/ticker/internal/ui/start.go:14 +0x1a5
github.com/achannarasappa/ticker/internal/cli.Run.func1(0xd00c40, 0xd3be70, 0x0, 0x0)
	/home/runner/work/ticker/ticker/internal/cli/cli.go:36 +0x2a
github.com/spf13/cobra.(*Command).execute(0xd00c40, 0xc00001e1d0, 0x0, 0x0, 0xd00c40, 0xc00001e1d0)
	/home/runner/go/pkg/mod/github.com/spf13/cobra@v1.1.1/command.go:854 +0x2c2
github.com/spf13/cobra.(*Command).ExecuteC(0xd00c40, 0x44570a, 0xcb7dc0, 0xc000042778)
	/home/runner/go/pkg/mod/github.com/spf13/cobra@v1.1.1/command.go:958 +0x375
github.com/spf13/cobra.(*Command).Execute(...)
	/home/runner/go/pkg/mod/github.com/spf13/cobra@v1.1.1/command.go:895
github.com/achannarasappa/ticker/cmd.Execute()


```

**Comment by achannarasappa** [comment]
It might be caused by the width of each cell being negative for some reason during a window resize event. I added handling for this - have a look at the latest version and let me know if that fixed it.

**Comment by viperperidot** [comment]
Unfortunately this update did not fix it..

**Comment by achannarasappa** [comment]
Any chance you'd be open to trying it in a VM? This would help isolate the issue. If the issue occurs in the VM that great then I can try replicating it on my end with the same setup. If not then it's possible it's a hardware issue of some kind.

Since I can not replicate the issue, it's any fix I attempt would be a shot in a dark.

**Comment by viperperidot** [comment]
I am not using a VM, but I can say that it was introduced in the previous version, I have been able to resize the window successfully prior to this.

**Comment by achannarasappa** [comment]
Would you be open to sharing your OS + version and terminal + version?

**Comment by viperperidot** [comment]
Ubuntu 20.04.2 LTS

GNOME Terminal 3.36.2

**Comment by MisterCubanSandwich** [comment]
I'm seeing similar behavior in macOS 10.15.7, zsh 5.7.1. If I try to launch ticker at my default size of 120x40, it throws that out of range error, but if I resize my terminal so it's not so wide it works correctly.

120 columns seems to be the threshold where it's broken for me.

**Comment by achannarasappa** [comment]
@MisterCubanSandwich do you are saying if the terminal window exceeds 120 columns the error is thrown? @viperperidot does resizing below 120 also work okay for you?

**Comment by MisterCubanSandwich** [comment]
Correct, it immediately crashes if launched or resized to wider than 120. I can resize it just fine if I keep it narrower than that.

**Comment by achannarasappa** [comment]
@MisterCubanSandwich would you mind sharing the error message? It would be helpful to understand if it is the exact same issue or a variant

**Comment by viperperidot** [comment]
@achannarasappa I just tested my threshold and for me I get the error kicking in at 105 column width.

**Comment by achannarasappa** [comment]
Thanks @viperperidot and @MisterCubanSandwich this info is really helpful. I'll take a shot at replicating the error

**Comment by achannarasappa** [comment]
Thanks for the details @viperperidot and @MisterCubanSandwich. I've put out a fix in release v4.0.4 that I believe address the issue. The root cause was that the width of the fixed width cells sometimes exceeded with width of the row which cause the error both of you observed. I've added better handling for this condition now and hopefully it will work as expected. When you have a chance take a look and let me know if the issue still occurs (and if it does please reshare the new error).

**Comment by achannarasappa** [comment]
Good news is I can now replicate the bug, bad news news it that the bug is not fixed :)

It seems to occur on window resize frequently enough to be annoying but not on every resize. I'll keep at a fix

**Comment by spsx** [comment]
I was about to report this bug (got ticker today) and it looks like you just patched this 20 minutes ago in ac6e26d2be6a4cc47cf1a18c371bfa7360f3fa63 :+1: . Picked up the latest master from git and it's now resizing properly with no crashes on my end.

**Comment by achannarasappa** [comment]
Thanks for the comment @spsx ! I'll close this issue but do reopen if the issue still resurfaces
